### ISSUE TITLE
Doesn't compile

### ISSUE THREAD

**Issue by BertilJ** [mannkind/ESPHomeRoombaComponent#3]
Stubled on this when looking for ways to automate my old vacuum. Great work! :)

But I'm getting a bunch of errors while compiling this, I wish I was able to fix these myself, but I don't have the competence for that.

```
In file included from src/main.cpp:12:0:
src/ESPHomeRoombaComponent.h:3:17: error: 'esphomelib' is not a namespace-name
using namespace esphomelib;
^
src/ESPHomeRoombaComponent.h:3:27: error: expected namespace-name before ';' token
using namespace esphomelib;
^
src/ESPHomeRoombaComponent.h:118:37: error: 'sensor' has not been declared
class RoombaDistanceSensor : public sensor::Sensor
^
src/ESPHomeRoombaComponent.h:118:45: error: expected '{' before 'Sensor'
class RoombaDistanceSensor : public sensor::Sensor
^
src/ESPHomeRoombaComponent.h:120:3: error: expected primary-expression before 'public'
public:
^
src/ESPHomeRoombaComponent.h:120:3: error: expected '}' before 'public'
src/ESPHomeRoombaComponent.h:120:3: error: expected ',' or ';' before 'public'
src/ESPHomeRoombaComponent.h:122:39: error: virt-specifiers in 'unit_of_measurement' not allowed outside a class definition
std::string unit_of_measurement() override { return "mm"; }
^
src/ESPHomeRoombaComponent.h:123:32: error: virt-specifiers in 'accuracy_decimals' not allowed outside a class definition
int8_t accuracy_decimals() override { return 1; }
^
src/ESPHomeRoombaComponent.h:124:1: error: expected declaration before '}' token
};
^
*** [/data/esp_roomba/.pioenvs/esp_roomba/src/main.cpp.o] Error 1
========================== [ERROR] Took 21.25 seconds ==========================
```

Would be really awesome if you could give me a pointer here!!

**Comment by mannkind** [comment]
Boo. Sorry about that. I'll take a look. 

My Roomba has been out of decommission for about a month (needs a part), so I cannot promise it all still works perfectly. But it should _at least_ compile.

**Comment by BertilJ** [comment]
Any effort is appreciated! I've got it working with a simpler solution covering only start/stop and go to base. But as I've got the rest of my home brewn iot in esphome would be cool to get my old roomba dancing again.

**Comment by mannkind** [comment]
Phew! It turns out that not only did ESPHome change the way custom components worked, but it appears that Home Assistant created a secondary schema ('state' vs 'legacy') for MQTT Vacuums.

I've attempted to update this component for both the 1.13.x series of ESPHome and the 'state' series of MQTT Vacuums in Home Assistant. The barebones _should_ be working again. I was able to cobble enough of my Roomba together to verify stats are flowing via MQTT but I cannot promise everything works 100%.

Stats:
```
roomba_001/status online
roomba_001/sensor/roomba_001_distance/state 0
roomba_001/sensor/roomba_001_voltage/state 18369
roomba_001/sensor/roomba_001_capacity/state 1348
roomba_001/sensor/roomba_001_current/state 1141
roomba_001/sensor/roomba_001_charge/state 272
roomba_001/binary_sensor/roomba_001_charging/state ON
roomba_001/binary_sensor/roomba_001_docked/state ON
```

UI:
<img width="367" alt="Screen Shot 2019-10-27 at 8 11 34 PM" src="https://user-images.githubusercontent.com/41883/67649516-a11a2800-f8f6-11e9-9b49-995c8396128e.png">


@BertilJ - Perhaps you could very that it's not completely messed up now?

**Comment by BertilJ** [comment]
Cool, I'll try to do some tests asap. In reality in a few days.

**Comment by BertilJ** [comment]
I've been trying to get this to work, since it is compiling for you I'm guessing I'm doing something wrong here, but I can figure out what.

```
In file included from src/main.cpp:30:0:
src/ESPHomeRoombaComponent.h:2:20: fatal error: roomba.h: No such file or directory

****************************************************************
* Looking for roomba.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:roomba.h"
* Web  > https://platformio.org/lib/search?query=header:roomba.h
*
****************************************************************

#include <roomba.h>
^
compilation terminated.
*** [/data/esp_roomba/.pioenvs/esp_roomba/src/main.cpp.o] Error 1
```

I'm seeing a new error message now, still won't compile fully. I've tried some minor changes here and there, but I can't get it to find roomba.h.

**Comment by mannkind** [comment]
Ack. the Roomba.yaml file is setup such that platformio  should download the required library from GitHub you. 

In a fresh container when I run `esphome roomba.yaml compile`, in the scrolling output I see something like...

```
LibraryManager: Installing Roomba
git version 2.20.1
Cloning into '/workspaces/ESPHomeRoombaComponent/example/roomba_001/.piolibdeps/roomba_001/_tmp_installing-ktCCif-package'...
Roomba @ dfe5975 has been successfully installed!
```

What version of esphome and PlatformIO are you using? do you have git installed?

**Comment by BertilJ** [comment]
Yeah, I was hoping the same! :)

I'm not getting that final line about successful installation of Roomba:

```
LibraryManager: Installing Roomba
git version 2.17.1
Cloning into '/data/esp_roomba/.piolibdeps/_tmp_installing-zmmBAf-package'...
Verbose mode can be enabled via `-v, --verbose` option
```
The Verbose line isn't connected to this, but I wanted to show where it ends. I'm using the same yaml-file, just changed the input for mqtt and wifi. 

Looks like I'm running another version of git than you do, ESPHome version is 1.13.6  and I don't have a clue what version of PlatformIO on top of that I'm running ESPHome as a Hass.io add-on, which means that I don't have direct access to the container. But checking the log, it looks like the compiling is started with a slightly  different command: `esphome --dashboard /config/esphome/esp_roomba.yaml compile` 

Don't know if this may have an impact, most probably this has to do with Hass.io add-on or docker installation.

**Comment by mannkind** [comment]
Ah-ha! Mentioning Hassio made me try compiling directly on my HA system...

Change line 2 in ESPHomeRoombaComponent.h from `#include <roomba.h>` to `#include <Roomba.h>`. 

I bet it'll compile then. If that works for you, I'll update the repo.

**Comment by BertilJ** [comment]
Magic moment! It did compile!! :) (after I changed the board to ESP8266) I was using ESP32, since that is what I had lying around. I guess I need to find another board to flash this with.

Many thanks!!

**Comment by mannkind** [comment]
Fixed in 169dd69.

No promises that it's perfect, but it's compiling and "working" so I'm closing this one for now.

**Comment by BertilJ** [comment]
Just an update, I finally got the time to play around some and all works nicely.
